fls-shake.nix

```
@@ -0,0 +1,27 @@
+{ pkgs ? import <nixpkgs> {} }:
+
+let
+  ghcEnv = pkgs.haskellPackages.ghcWithPackages (hp: [
```

```suggestion
  ghc = pkgs.haskellPackages.ghcWithPackages (hp: [
```
Rename this to just `ghc`  (I don't like the suffix `Env`)

---

Actually, I don't like using `ghc` for the environment because its too easy to confuse it with the `ghc` command (which is used later, in line 20).